app/models/paper.rb

```diff
@@ -43,4 +45,10 @@ def valid_methodologies
       'case report'
     ]
   end
+
+  def find_publication
+    if self.publication.present?
+      self.publication = Publication.where(name: self.publication.name).first_or_initialize
```

Comment:
Once I took out the validate_uniqueness from publication.rb, I think we
prob don't need to do it before_validation?  Where's the best place to put
it if we don't need it before validation?

On Tue, Aug 23, 2016 at 2:58 PM, Brian Kung <notifications@github.com>
wrote:

> In app/models/paper.rb
> <https://github.com/briankung/projectcredo/pull/64#discussion_r75939090>:
>
> > @@ -43,4 +45,10 @@ def valid_methodologies
> >        'case report'
> >      ]
> >    end
> > +
> > +  def find_publication
> > +    if self.publication.present?
> > +      self.publication = Publication.where(name: self.publication.name).first_or_initialize
>
> This whole before_validation hook seems very wrong. Why is this necessary?
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/briankung/projectcredo/pull/64/files/c10ff7dd636d241cfd3c03b300def8df2400222c#r75939090>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/ACKsZBm91QW9AoxQqMK5kQw9BN6a8TSBks5qi1DhgaJpZM4JqR56>
> .
>